src/index.js

```diff
@@ -6,174 +6,139 @@
-        body: 'Not Found'
-      });
-    } else if (err instanceof this.resource.errorClass) {
-      this.respond(null, {
-        statusCode: 400,
-        headers: { 'Content-Type': 'text/plain' },
-        body: err.toString()
-      });
-    } else {
-      this.respond(err, null);
-    }
-  }
+const handleRequestFunc = async (event, context, callback) => {
+  AWS.config.region = context.invokedFunctionArn.match(/^arn:aws:lambda:(\w+-\w+-\d+):/)[1];
 
-  includeStage() {
-    if ('include_stage' in process.env) {
-      return ['true', 'yes'].indexOf(process.env.include_stage.toLowerCase()) > -1;
-    } else {
-      var host = this.event.headers['Host'];
-      return host.match(/\.execute-api\.\w+?-\w+?-\d+?\.amazonaws\.com$/);
-    }
+  // OPTIONS REQUEST
+  if (event.httpMethod === 'OPTIONS') {
+    console.log('OPTIONS REQUEST');
+    await callback(null, { statusCode: 204, body: null });
   }
-
-  fileMissing() {
-    return !/\.(jpg|tif|gif|png|json)$/.test(this.event.path);
+  // INFO.JSON REQUEST
+  else if (fileMissing(event)) {
+    const location = eventPath(event) + '/info.json';
+    console.log(`INFO.JSON REQUEST: ${location}`);
+    await callback(null, { statusCode: 302, headers: { 'Location': location }, body: 'Redirecting to info.json' });
   }
-
-  eventPath() {
-    if ('storedEventPath' in this) return this.storedEventPath;
-
-    var path = this.event.path;
-    if (this.includeStage()) {
-      path = '/' + this.event.requestContext.stage + path;
+  // IMAGE REQUEST
+  else {
+    const uri = getUri(event);
+    const resource = new IIIF.Processor(
+      uri,
+      streamResolver,
+      dimensionResolver
+    );
+    const result = await resource.execute();
+    try {
+      const base64 = isBase64(result)
+      const content = base64 ? result.body.toString('base64') : result.body;
+      const response = {
+        statusCode: 200,
+        headers: {
+          'Content-Type': result.contentType,
+          'Access-Control-Allow-Origin': '*'
+         },
+        isBase64Encoded: base64,
+        body: content
+      };
+      if (content.length > 6 * 1024 * 1024) {
+        const uri = getUri(event);
+        console.log(`base64.length = ${content.length}, original.length =  ${result.body.length}`);
+        throw Error (`Content size (${content.length.toString()}) exceeds API gateway maximum when calling ${uri}`);
+      } else {
+        await callback(null, response);
+      }
+    } catch (err) {
+      return errorHandler(err, event, context, resource, callback)
     }
-    this.storedEventPath = path.replace(/\/*$/, '');
-    return this.storedEventPath;
   }
+};
 
-  checkForOptionsRequest() {
-    if (this.handled) return this;
-    if (this.event.httpMethod === 'OPTIONS') {
-      this.respond(null, { statusCode: 204, body: null });
-      this.handled = true;
-    }
-    return this;
+const errorHandler = async (err, event, context, resource, callback) => {
+  console.error(err);
```

Comment:
console?

Reply:
It was in the original code. I think it shows up in the logs.